### ISSUE TITLE
Backup not working

### ISSUE THREAD

**Issue by StarPicard** [signalapp/Signal-Android#8054]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When trying to enable the Backup it won't work in any way. I have tried granting all rights and temporarily unmounting the external SD-Card. Status of the last backup is always "unknown" (german: "unbekannt"). It sounds exaktly like the following: [https://www.reddit.com/r/signal/comments/8t58ss/backup_unable_to_be_made/](https://www.reddit.com/r/signal/comments/8t58ss/backup_unable_to_be_made/)

### Steps to reproduce
- Enable the Feature "Save Chats"
- Following the tool-tip steps (numbers and check)
- Manually clicking on "Create Backup"
- Signal Icon flashing in the bar and instantly disappear

**Actual result:** Nothing happens
**Expected result:** Backup should be created

### Device info

**Device:** Motorola Moto G3 (osprey)
**Android version:** 7.1.2
**Signal version:** 4.23.4 (Non-Google: manual APK-Download)

### Link to debug log
```
Device  : Motorola MotoG3 (lineage_osprey)
Android : 7.1.2 (e037f4a905, lineage_osprey-userdebug 7.1.2 NJH47F e037f4a905)
Memory  : 33M (25.30% free, 512M max)
Memclass: 192
OS Host : agrippa.acc.umu.se
App     : Signal 4.23.4

--------- beginning of main
07-30 19:12:56.640  6290  6290 W BackupUtil: org.thoughtcrime.securesms.database.NoExternalStorageException
07-30 19:12:56.640  6290  6290 W BackupUtil:    at org.thoughtcrime.securesms.util.StorageUtil.getBackupDirectory(StorageUtil.java:21)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at org.thoughtcrime.securesms.util.BackupUtil.getLatestBackup(BackupUtil.java:36)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at org.thoughtcrime.securesms.util.BackupUtil.getLastBackupTime(BackupUtil.java:25)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at org.thoughtcrime.securesms.preferences.ChatsPreferenceFragment.setBackupSummary(ChatsPreferenceFragment.java:111)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at org.thoughtcrime.securesms.preferences.ChatsPreferenceFragment.onResume(ChatsPreferenceFragment.java:80)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.support.v4.app.Fragment.performResume(Fragment.java:2308)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1465)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1750)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1819)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:797)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2590)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2377)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.support.v4.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2332)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2239)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:700)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.os.Handler.handleCallback(Handler.java:751)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.os.Handler.dispatchMessage(Handler.java:95)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.os.Looper.loop(Looper.java:154)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at android.app.ActivityThread.main(ActivityThread.java:6186)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at java.lang.reflect.Method.invoke(Native Method)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
07-30 19:12:56.640  6290  6290 W BackupUtil:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
07-30 19:12:59.201  6290  6290 W ChatsPreferenceFragment: Queing backup...
07-30 19:12:59.204  6290  6311 W LocalBackupJob: Executing backup job...
07-30 19:12:59.214  6290  6311 W JobConsumer: 
07-30 19:12:59.214  6290  6311 W JobConsumer: org.thoughtcrime.securesms.database.NoExternalStorageException
07-30 19:12:59.214  6290  6311 W JobConsumer:   at org.thoughtcrime.securesms.util.StorageUtil.getBackupDirectory(StorageUtil.java:21)
07-30 19:12:59.214  6290  6311 W JobConsumer:   at org.thoughtcrime.securesms.jobs.LocalBackupJob.onRun(LocalBackupJob.java:55)
07-30 19:12:59.214  6290  6311 W JobConsumer:   at org.thoughtcrime.securesms.jobmanager.JobConsumer.runJob(JobConsumer.java:74)
07-30 19:12:59.214  6290  6311 W JobConsumer:   at org.thoughtcrime.securesms.jobmanager.JobConsumer.run(JobConsumer.java:47)
```

**Comment by greyson-signal** [comment]
So it looks like Signal is unable to write to your external storage directory. To be clear, "external storage" does not necessarily mean an external SD card. Android has kind of overloaded the term, and in fact it typically refers to internal storage. That being said, it could really be anywhere the phone specifies it to be. The important thing here is that we're being told that it isn't writable.

This is really the only reasonable place for us to look to write backups, as it's the only way the file can survive uninstall/reinstall. So if you're phone is reporting it can't be written to, I don't know if there's any recourse from our side.

I'd check the following:
- Are you using a rooted device or a non-stock ROM? If so, there are apps people have installed in the past (like Xposed) that have messed with our storage permissions.
- Is your phone mounted to a computer? Looking at the docs, storage may not be writable during that time.
- Make sure you're rebooting your phone in between attempts of removing/inserting your external SD card.

**Comment by StarPicard** [comment]
Thanks, for the reply. I am using lineage (custom rom) with root. So it really seems to be "my" problem.
Sadly I don't know any app which could have caused such an behavior. I'll take a look at it and give feedback if I found a solution.

I only have one suggestion regarding the message in signal. Won't it be a nice solution to display a text like "Unable to write at local storage."?

**Comment by StarPicard** [comment]
I've done some research but did not find anything useful. Is there any investigation point you can give me?

**Comment by StarPicard** [comment]
Okay, excuse me! Simple but stupid error. Somehow a hidden privacy feature in my lineage install didn't allow Signal to read the external storage even when the permission was granted. I've found it while trying to export an image from the app. When I tried this a message was shown, that Signal has no rights to write/read external storage. 

My suggestion from above still is important in my eyes. The same error message as trying to save images should be given when signal can't access external storage for backups.
